### ISSUE TITLE
Warning: Accessing non-existent property 'MongoError' of module exports inside circular dependency

### ISSUE THREAD

**Issue by orassayag** [Automattic/mongoose#9966]
**Do you want to request a *feature* or report a *bug*?**

**What is the current behavior?**
After upgrade to mongoose 5.11.18 I'm getting a warning:
"(node:16924) Warning: Accessing non-existent property 'MongoError' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)"

**If the current behavior is a bug, please provide the steps to reproduce.**

Upgrade to mongoose 5.11.18 and try to use it.

**What is the expected behavior?**

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
Using Node v14.15.5, mongoose 5.11.18

**Comment by WickedWizard3588** [comment]
Hey,

I have the same problem, node version `v15.10.0` and mongoose version, `5.11.18`. This occurs when I try to search for a database that does not exist.

**Comment by sempasha** [comment]
This warning appears for mongoose `^5.11.16`, when `5.11.15` installed there is no warning.

**Comment by WickedWizard3588** [comment]
It happens on 5.11.15 for me.

**Comment by thomasthebro1** [comment]
I am also experiencing this problem with `5.11.18`, best bet is probably to revert to `5.11.15` until the problem is fixed as it results in the database not connecting at all.

**Comment by vkarpov15** [comment]
This is a duplicate of #9900. The fix has been merged https://github.com/mongodb/node-mongodb-native/pull/2734 but we're still waiting on a new release of the [MongoDB driver](https://www.npmjs.com/package/mongodb) in order to pull in that fix.
